### ISSUE TITLE
--no-derive-debug prevents even manual Debug on enumerations

### ISSUE THREAD

**Issue by VeaaC** [rust-lang/rust-bindgen#2076]
### Input C/C++ Header

```C++
// lots of enums like this
/// <div rustbindgen derive="Clone"></div>
/// <div rustbindgen derive="Debug"></div>
/// <div rustbindgen derive="Copy"></div>
/// <div rustbindgen derive="PartialEq"></div>
/// <div rustbindgen derive="PartialOrd"></div>
enum NeedsDebug { VALUE };

// Lots of structs like this
typedef struct MustNotHaveDebug MustNotHaveDebug;
```

### Bindgen Invocation

```
bindgen --no-derive-debug --rustified-enum input.h
```

### Actual Results

```rust
/* automatically generated by rust-bindgen 0.59.0 */

#[repr(u32)]
#[doc = " <div rustbindgen derive=\"Clone\"></div>"]
#[doc = " <div rustbindgen derive=\"Debug\"></div>"]
#[doc = " <div rustbindgen derive=\"Copy\"></div>"]
#[doc = " <div rustbindgen derive=\"PartialEq\"></div>"]
#[doc = " <div rustbindgen derive=\"PartialOrd\"></div>"]
#[derive(Copy, Clone, Hash, PartialEq, Eq)]
pub enum NeedsDebug {
    VALUE = 0,
}
#[repr(C)]
#[derive(Copy, Clone)]
pub struct MustNotHaveDebug {
    _unused: [u8; 0],
}

```

### Expected Results

I would expect manual overrides to overrule default settings. The main use case here is that I need Debug on all enumerations, a few structs, but not on most other types.

**Comment by emilio** [comment]
curious, does `Default` behave like `Debug`?

**Comment by VeaaC** [comment]
@emilio it seems that Debug is the only thing behaving like this:

```
bindgen --no-derive-debug --no-derive-default --rustified-enum .\* input.h 

/* automatically generated by rust-bindgen 0.59.0 */

#[repr(u32)]
#[doc = " <div rustbindgen derive=Clone></div>"]
#[doc = " <div rustbindgen derive=Debug></div>"]
#[doc = " <div rustbindgen derive=Copy></div>"]
#[doc = " <div rustbindgen derive=Default></div>"]
#[doc = " <div rustbindgen derive=PartialEq></div>"]
#[doc = " <div rustbindgen derive=PartialOrd></div>"]
#[derive(Copy, Clone, Hash, PartialEq, Eq)]
pub enum NeedsDebug {
    VALUE = 0,
}
#[repr(C)]
#[derive(Copy, Clone)]
pub struct MustNotHaveDebug {
    _unused: [u8; 0],
}
```

The problem is also a recent change in behaviour: In version 0.57 enumerations would get Debug independently of whether  `--no-derive-debug` was specified or not (which accidentally suited my use-case well.

**Comment by VeaaC** [comment]
I realized that enumeration are not adding any `derive` directives from annotations at all. It is a simple change to support that, e.g.:

``` diff
@@ -3003,13 +3004,16 @@ impl CodeGenerator for Enum {
             // For backwards compat, enums always derive Clone/Eq/PartialEq/Hash, even
             // if we don't generate those by default.
             derives.insert(
-                DerivableTraits::CLONE |
-                    DerivableTraits::COPY |
-                    DerivableTraits::HASH |
-                    DerivableTraits::PARTIAL_EQ |
-                    DerivableTraits::EQ,
+                DerivableTraits::CLONE
+                    | DerivableTraits::COPY
+                    | DerivableTraits::HASH
+                    | DerivableTraits::PARTIAL_EQ
+                    | DerivableTraits::EQ,
+            );
+            let mut derives: Vec<_> = derives.into();
+            derives.extend(
+                item.annotations().derives().iter().map(String::as_str),
             );
-            let derives: Vec<_> = derives.into();
             attrs.push(attributes::derives(&derives));
```

However, the code also talks about backwards compatibility:
```
            // For backwards compat, enums always derive Clone/Eq/PartialEq/Hash, even
            // if we don't generate those by default.
```

And it seems like `Debug` should have most likely also been part of the same backwards compatibility.

I could prepare a PR with a fix / implementation based on what kind of change you would prefer (supporting annotations, adding Debug always, or both).

**Comment by VeaaC** [comment]
I created two possible fixes for this: #2089 is kind of a minimal fix, restoring the previous behaviour. #2090 is a more complete fix, restoring the previous behaviour, but also adding full support for annotations on enumerations (which allows the user to opt out of Debug)
